docs/RFCS/20171118_role_based_access_control.md

```diff
@@ -0,0 +1,757 @@
+- Feature Name: Role-based access control
+- Status: draft
+- Start Date: 2017-11-18
+- Authors: Marc Berhault
+- RFC PR: [#20149](https://github.com/cockroachdb/cockroach/pull/20149)
+- Cockroach Issue: (one or more # from the issue tracker)
+
+
+Table of Contents
+=================
+
+   * [Summary](#summary)
+   * [Motivation](#motivation)
+   * [Related resources](#related-resources)
+   * [Out of scope](#out-of-scope)
+   * [Guide-level explanation](#guide-level-explanation)
+      * [Terminology](#terminology)
+      * [User-level explanation](#user-level-explanation)
+         * [Role basics](#role-basics)
+         * [Creating a role](#creating-a-role)
+         * [Granting and revoking privileges for a role](#granting-and-revoking-privileges-for-a-role)
+         * [Manipulating role membership](#manipulating-role-membership)
+         * [Dropping a role](#dropping-a-role)
+         * [Listing roles](#listing-roles)
+         * [Listing role memberships](#listing-role-memberships)
+         * [Administrators](#administrators)
+         * [Non-enterprise users](#non-enterprise-users)
+      * [Contributor impact](#contributor-impact)
+   * [Reference-level explanation](#reference-level-explanation)
+      * [Detailed design](#detailed-design)
+         * [Roles system table](#roles-system-table)
+         * [New SQL statements](#new-sql-statements)
+         * [Checking permissions](#checking-permissions)
+         * [Determining role memberships](#determining-role-memberships)
+         * [Admin role](#admin-role)
+         * [Migrations and backwards compatibility](#migrations-and-backwards-compatibility)
+         * [Non-enterprise functionality](#non-enterprise-functionality)
+         * [Virtual tables](#virtual-tables)
+      * [Drawbacks](#drawbacks)
+      * [Rationale and Alternatives](#rationale-and-alternatives)
+      * [Unresolved questions](#unresolved-questions)
+      * [Future improvements](#future-improvements)
+
+# Summary
+
+Role-based access control is an enterprise feature.
+
+We propose the addition of SQL roles to facilitate user management.
+
+A role can be seen as a group containing any number of "entities" as members. An entity can be a user or
+another role.
```

Comment:
there can be no cycles in this relationship

Reply:
This is mentioned in the set of rules in `Role basics`